hammer/vcs/git.py

```diff
@@ -184,6 +184,10 @@ def get_branch(self, commit_id='HEAD', ambiguous=False):
         with cd(self.code_dir), hide('running'):
             valid_branches, real_commit = self._get_commit_branch(commit_id)
 
+            # Use the branch cache, if possible.
+            if real_commit and real_commit in self._branch_cache:
```

Comment:
I moved the branch cache above the _get_commit_branch call to avoid calling it, when possible. Note that _get_commit_branch is not called anywhere else, so there is no need to duplicate this logic.